tao_compiler/mlir/xla/ral/context/common_context_impl_quantization.cc

```diff
@@ -353,6 +353,316 @@ void ral_qconv_acl_s8_s8_s8_per_channel(
   }
 }
 
+template <int NDims>
+MemRefType<int8_t, NDims> ral_pdll_qconv_acl_s8_s8_s8_per_channel(
+    ExecutionContext* ctx, opaque_t /*stream_handle*/,
+    MemRefType<int8_t, NDims> input, MemRefType<int8_t, NDims> weight,
+    MemRefType<float, 0> inputScales, MemRefType<int32_t, 0> inputZeroPoints,
+    MemRefType<float, 1> weightScales, MemRefType<int32_t, 1> weightZeroPoints,
+    MemRefType<float, 0> resultScales, MemRefType<int32_t, 0> resultZeroPoints,
+    void* customAttrs) {
+  CpuTimer timer("ral_pdll_qconv");
+  TAO_VLOG(0) << "GeePL Test\n";
+  int64_t resultSizes[NDims] = {0};
+  if (isEmptyMemref(input) || isEmptyMemref(weight)) {
+    TAO_VLOG(1) << "ral_pdll_qconv_acl_s8_s8_s8_per_channel: early return for "
+                   "empty tensor";
+    return assignMemRef<int8_t, NDims>(nullptr, resultSizes);
+  }
+
+  auto attr = getOrParsePDLAttr(ctx, customAttrs,
+                                "ral_pdll_qgemm_acl_s8_s8_s8_per_channel");
+  if (!attr) {
+    ctx->signalError(Context::FAILURE, "fail to parse custom_attrs\n");
+  }
+
+  auto& dictAttr = attr->as<DictPDLAttr>();
+  auto& data_format_str =
```

Comment:
please put these into a helper function since other users (e.g. conv2d+biasadd) also need this.